File: task9/task_9.cpp
#include <vector>

using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0 || x % 10 == 0 && x != 0) {
            return false;
        }

        int revertedNumber = 0;
        while (x > revertedNumber) {
            revertedNumber = revertedNumber * 10 + x % 10;
            x /= 10;
        }

        return revertedNumber == x || x == revertedNumber / 10;
    }
};

File: task9/task_9.go
package main

func isPalindrome(x int) bool {
    if x < 0 || x % 10 == 0 && x != 0 {
		return false
	}

	reversedNumber := 0
	for ; x > reversedNumber; {
		reversedNumber = reversedNumber * 10 + x % 10
		x /= 10
	}

	return reversedNumber == x || x == reversedNumber / 10
}